common/schema-builder/src/main/java/io/smallrye/graphql/schema/creator/DirectiveTypeCreator.java

```diff
@@ -79,8 +79,8 @@ public DirectiveType create(ClassInfo classInfo) {
                 argumentType = method.returnType();
             }
             argument.setReference(referenceCreator.createReferenceForOperationArgument(argumentType, null));
-            argument.setName(method.name());
             Annotations annotationsForMethod = Annotations.getAnnotationsForInterfaceField(method);
+            argument.setName(TypeNameHelper.getMethodName(method, annotationsForMethod));
```

Comment:
This is done due to `io.smallrye.graphql.api.federation.link.Link`, as we cannot use 'import' and 'for' as variable names in Java.
```java
@Name("import")
Import[] _import() default {};

@Name("for")
Purpose _for() default Purpose.UNDEFINED;
```
```graphql
directive @link(url: String!, as: String, import: [Import], for: Purpose) repeatable on SCHEMA
```